app/routes/bdb/components/BdbDetail.tsx

```diff
@@ -41,27 +47,91 @@ import {
 import { selectAllCompanySemesters } from 'app/reducers/companySemesters';
 import { selectPaginationNext } from 'app/reducers/selectors';
 import { selectUserById } from 'app/reducers/users';
+import SemesterStatus from 'app/routes/bdb/components/SemesterStatus';
+import {
+  indexToCompanySemester,
+  semesterToHumanReadable,
+} from 'app/routes/bdb/utils';
 import { displayNameForEventType } from 'app/routes/events/utils';
 import { useAppDispatch, useAppSelector } from 'app/store/hooks';
 import { EntityType } from 'app/store/models/entities';
 import truncateString from 'app/utils/truncateString';
-import { sortByYearThenSemester, getContactStatuses } from '../utils';
-import SemesterStatusDetail from './SemesterStatusDetail';
 import styles from './bdb.css';
+import type { EntityId } from '@reduxjs/toolkit';
+import type { ColumnProps } from 'app/components/Table';
 import type { TransformedSemesterStatus } from 'app/reducers/companies';
 import type {
+  CompanyContact,
   CompanySemesterContactStatus,
-  SemesterStatus,
 } from 'app/store/models/Company';
 import type { ListEvent } from 'app/store/models/Event';
 
+type RenderFileProps = {
+  semesterStatus: TransformedSemesterStatus;
+  type: string;
+  removeFile: (
+    type: string,
+    semesterStatus: TransformedSemesterStatus,
+  ) => Promise<unknown>;
+  addFile: (
+    name: string,
+    token: string,
+    type: string,
+    semesterStatus: TransformedSemesterStatus,
+  ) => Promise<unknown>;
+};
+
+export const RenderFile = (props: RenderFileProps) => {
+  const { semesterStatus, type, removeFile, addFile } = props;
+
+  const fileNameToShow = (name: string, url?: string) =>
+    name ? <a href={url}>{truncateString(name, 30)}</a> : '-';
```

Comment:
Could the "-" have a secondary font color?😁
```suggestion
    name ? <a href={url}>{truncateString(name, 30)}</a> : <span className="secondaryFontColor">-</span>;
```